CsCore/PlainNetClassLib/src/Plugins/CsCore/com/csutil/algorithms/AaaaImageCompareAaaa.cs

```diff
@@ -0,0 +1,107 @@
+﻿using StbImageSharp;
+using System;
+using System.Drawing;
+using System.Numerics;
+
+namespace com.csutil.src.Plugins.CsCore.com.csutil.algorithms {
+
+    public class AaaaImageCompareAaaa {
+
+        public AaaaImageCompareAaaa() { }
+
+        public CompareResult ImageCompare(ImageResult img1, ImageResult img2, double redScale=5) {
+
+            CompareResult result = new CompareResult();
+
+            ImageResult resultImage = new ImageResult();
+            resultImage.Data = img1.Data.DeepCopy();
+            resultImage.Width = img1.Width;
+            resultImage.Height = img1.Height; 
+
+            // Check if images are the same size
+            if (img1.Width != img2.Width && img1.Height != img2.Height) {
+                result.resultImage = resultImage;
+                return result;
+            }
+
+            // Load pixel colors
+            Color[] colors_img1 = LoadImageColors(img1);
+            Color[] colors_img2 = LoadImageColors(img2);
+
+            // Calculate distortion
+            result.distortion = GetMeanSquaredDistortion(colors_img1, colors_img2, resultImage, redScale);
+            result.resultImage = resultImage;
+
+            return result;
+
+        }
+
+        private Color[] LoadImageColors(ImageResult img) {
```

Comment:
would it be possible to change this to directly return a 2d array here? This method then could become a nice extension method in the ImageResultExtensions.cs to read the image content out as a 2d array of colors (maybe then consider Color to be renamed to Pixel?)

Reply:
maybe then the resulting Color[][] could also be wrapped into a wrapper class that you call ImageResultV2 or something like this in case there will be more fields in that class in addition to the Color[][] array

Reply:
See also related comment https://github.com/cs-util-com/cscore/pull/128/files#r1505378988